supervision/metrics/detection.py

```diff
@@ -0,0 +1,420 @@
+        ...             ]
+        ...         ),
+        ...         array([1.0, 1.0, 2.0, 2.0, 2]),
+        ...     ]
+        ... )
+        ...
+        >>> predictions = [
+        ...     array(
+        ...         [
+        ...             [0.0, 0.0, 3.0, 3.0, 1, 0.9],
+        ...             [0.1, 0.1, 3.0, 3.0, 0, 0.9],
+        ...             [6.0, 1.0, 8.0, 3.0, 1, 0.8],
+        ...             [1.0, 6.0, 2.0, 7.0, 1, 0.8],
+        ...         ]
+        ...     ),
+        ...     array([[1.0, 1.0, 2.0, 2.0, 2, 0.8]])
+        ... ]
+
+        >>> confusion_matrix = ConfusionMatrix.from_tensors(
+        ...     predictions=predictions,
+        ...     targets=targets,
+        ...     num_classes=3
+        ... )
+
+        >>> confusion_matrix.matrix
+        ... array([
+        ...     [0., 0., 0., 0.],
+        ...     [0., 1., 0., 1.],
+        ...     [0., 1., 1., 0.],
+        ...     [1., 1., 0., 0.]
+        ... ])
+        ```
+        Source: https://github.com/SkalskiP/onemetric/blob/master/onemetric/cv/object_detection/confusion_matrix.py
+        """
+        cls._validate_input_tensors(predictions, targets)
+
+        num_classes = len(classes)
+        matrix = np.zeros((num_classes + 1, num_classes + 1))
+        for true_batch, detection_batch in zip(targets, predictions):
+            matrix += ConfusionMatrix._evaluate_detection_batch(
+                true_detections=true_batch,
+                pred_detections=detection_batch,
+                num_classes=num_classes,
+                conf_threshold=conf_threshold,
+                iou_threshold=iou_threshold,
+            )
+        return cls(
+            matrix=matrix,
+            classes=classes,
+            conf_threshold=conf_threshold,
+            iou_threshold=iou_threshold,
+        )
+
+    @classmethod
+    def _validate_input_tensors(cls, predictions, targets):
+        """Checks for shape consistency of input tensors."""
+        if len(predictions) != len(targets):
+            raise ValueError(
+                f"Number of predictions ({len(predictions)}) and targets ({len(targets)}) must be equal."
+            )
+        if len(predictions) > 0:
+            if not isinstance(predictions[0], np.ndarray) or not isinstance(
+                targets[0], np.ndarray
+            ):
+                raise ValueError(
+                    f"Predictions and targets must be lists of numpy arrays. Got {type(predictions[0])} and {type(targets[0])} instead."
+                )
+            if predictions[0].shape[1] != 6:
+                raise ValueError(
+                    f"Predictions must have shape (N, 6). Got {predictions[0].shape} instead."
+                )
+            if targets[0].shape[1] != 5:
+                raise ValueError(
+                    f"Targets must have shape (N, 5). Got {targets[0].shape} instead."
+                )
+
+    @staticmethod
+    def _evaluate_detection_batch(
+        true_detections: np.ndarray,
```

Comment:
Also let's use consistent naming conventions and name arguments: `predictions` and `targets`.